### PR TITLE
Adding use-map

### LINKED ISSUE #3: Feat: Map
The implementation of the [`Map` data structure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map).

### PR DESCRIPTION
**Pull request by wesleimp**
Fixes #3 

* [x] - Implementation ✔️ 
* [x] - Tests
* [ ] - Documentation / Code sample

### CODE REVIEW

**Comment by zaguiini** [comment]
That's great! For the tests, please use https://github.com/testing-library/react-hooks-testing-library
I'm creating the tests for the `Set` hook and will provide an example soon

File: src/use-map.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useState, useMemo } from 'react';
+
+export const useMap = <TKey, TValue>(initialState = new Map<TKey, TValue>()) => {
+  const [value, setValue] = useState(initialState);
+
+  const handlers = useMemo(() => Object.assign(value, {
+    set: (key: TKey, value: TValue) => {
+      setValue((oldMap: Map<TKey, TValue>): Map<TKey, TValue> => {
```
Comment (zaguiini) [inline src/use-map.ts]: I think you can omit the return type annotation here since the TS compiler can infer it

File: src/use-map.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    delete: (key: TKey) => {
+      setValue((oldMap) => {
+        const newMap = new Map(oldMap);
+        newMap.delete(key);
+
+        return newMap
+      })
+    },
+
+    clear: () => setValue(new Map()),
```
Comment (zaguiini) [inline src/use-map.ts]: Please leave the methods on alphabetic order

**Comment by zaguiini** [comment]
I'm thinking about changing the underlying implementation to use Immutable.JS. Thoughts on that?

**Comment by wesleimp** [comment]
> That's great!
> 
> Please, use [this test](https://github.com/zaguiini/use-data-structures/blob/master/test/use-set.test.ts) as guidance for the tests implementation!
> 
> Don't forget to export the hook on the `index.ts` file and remove the `Map` item from the TODO section on README

@zaguiini  yes, I was waiting for you to write the tests to follow the owner's tests pattern.

What do you think about remove TODO section from README and create issues with label for those ones? (Just thinking loud).

File: src/use-map.ts (hunk 2 above)
Comment (wesleimp) [reply src/use-map.ts]: Yeap

**Comment by zaguiini** [comment]
> What do you think about remove TODO section from README and create issues with label for those ones? (Just thinking loud).

That's a good idea.

File: src/use-map.ts (hunk 1 above)
Comment (wesleimp) [reply src/use-map.ts]: yes, no probs

**Comment by wesleimp** [comment]
> > What do you think about remove TODO section from README and create issues with label for those ones? (Just thinking loud).
> 
> That's a good idea. On it! Please rebase from `master` since there are changes.

Done ✔️

**Comment by wesleimp** [comment]
> All looks good. Missing the README changes including the usage. After approval and publishing I'll include the CodeSanbox link.

Sorry, my bad

**Pull request merged**